test/tests/amqp/test_amqp_node_rediscover.py

```diff
@@ -0,0 +1,412 @@
+            nodeid +
+            ".#",
+            external_callback=self.amqp_callback,
+            timeout=30)
+        td.setDaemon(True)
+        td.start()
+        result = fit_common.rackhdapi(
+            '/api/2.0/nodes/' + nodeid, action='delete')
+        self.assertTrue(result[
+                        'status'] < 209, 'Was expecting response code < 209. Got ' + str(result['status']))
+        self._wait_amqp_message(10)
+        amqp_message = amqp_queue.get()
+        self._process_message("removed", nodeid, nodeid, "node", amqp_message)
+        # start discovery
+        logs.debug_2("Waiting node reboot and boot into microkernel........")
+        td.dispose()
+        # clear the amqp message queue
+        while amqp_queue.empty is False:
+            amqp_queue.get()
+        logs.debug('launch AMQP thread for discovery')
+        discover_worker = AmqpWorker(
+            exchange_name="on.events",
+            topic_routing_key="node.#.information.#",
+            external_callback=self.amqp_callback,
+            timeout=600)
+        discover_worker.setDaemon(True)
+        discover_worker.start()
+        logs.debug('Wait for node added')
+        self.assertTrue(
+            self._wait_for_discover(node_uuid),
+            "Fail to find the orignial node after reboot!")
+        logs.debug_2(
+            "Found the orignial node. It is rediscovered succesfully!")
+        logs.debug('Wait for node discovery')
+        self._wait_amqp_message(60)
+        amqp_message_discover = amqp_queue.get()
+        self._process_message(
+            "discovered",
+            nodefound_id,
+            nodefound_id,
+            "node",
+            amqp_message_discover)
+        # skip sku.assigned message if no skupack is installed on the ora
+        if skupack_intalled:
+            logs.debug_2("wait for skupack assign!")
+            self._wait_amqp_message(50)
+            amqp_message_discover = amqp_queue.get()
+            self._process_message(
+                "sku.assigned",
+                nodefound_id,
+                nodefound_id,
+                "node",
+                amqp_message_discover)
+        logs.debug_3("wait for obm assign!")
+        # re-apply obm setting to the node to generate obm.assigned message
+        self.assertTrue(self._apply_obmsetting_to_node(
+            nodefound_id), "Fail to apply obm setting!")
+        self._wait_amqp_message(50)
+        amqp_message_discover = amqp_queue.get()
+        self._process_message(
+            "obms.assigned",
+            nodefound_id,
+            nodefound_id,
+            "node",
+            amqp_message_discover)
+        logs.debug_3("wait for accessbile!")
+        self._wait_amqp_message(100)
+        amqp_message_discover = amqp_queue.get()
+        self._process_message(
+            "accessible",
+            nodefound_id,
+            nodefound_id,
+            "node",
+            amqp_message_discover)

```

Comment:
Add the following code to cover RI-48:
amqp_body = amqp_message_discover[1];
assert  "nodeId" in amqp_body["data"], "nodeId is not contained in the discover message"
assert  amqp_body["data"]["nodeId"], "nodeId generated in discovery doesn't include valid data "
assert  "ipMacAddresses" in amqp_body["data"], "ipMacAddresses is not contained in the discover message"
assert amqp_body["data"]["ipMacAddresses"], " ipMacAddresses generated during node discovery doesn't include valid data ",